### DISCUSSION TITLE
EspHome-AwoX-BLE-mesh-hub setup

### DISCUSSION THREAD

**Discussion by MBInoob** [fsaris/home-assistant-awox#119] [Q&A]
As an HA noob, I tried my best to understand the [EspHome-AwoX-BLE-mesh-hub](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub) setup page:

1- Installed ESPHome as a Home Assistant add-on (quick access shows in the sidebar),

2- Picked an ESP32 module online ([WIFI & Bluetooth capable](https://www.amazon.fr/dp/B08BZGC22Q?)),

3- Ran a PC>USB>ESP32 install using HA (can't remember what I did exactly, but there was a 'secret' something with the WIFI password),

4- The ESP32 module seems connected as WIFI signal strength shows into the ESPHome logs while simply plugged into a powerstrip, and it also shows up as a device in the correct room,

5- Installed Mosquitto as a HA add-on because the AwoX mesh setup page mentioned MQTT (probably messed that one up, unsure what to do with this),

6- Pasted all the files from the [component](https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub) page into HOMMEASSISTANT>config>esphome folder using PC without changing anything to the files (unsure that's even needed),

7- Found my AwoX connection password to retrieve my mesh credentials, but unsure what to do with them.

Strangely, HA has added two of my AwoX EGLO bulbs as they show up in the "Overview", but the buttons cannot be activated (just like with the [home assistant awox](https://github.com/fsaris/home-assistant-awox) integration I first unsuccessfully tried - now removed). Notifications now state:

 "setup failed for 'awox': Integration not found" error setup.py 
"Failed to load integration for translation: Integration 'awox' not found" helpers/translation.py

Have restarted HA (running from a freshly purchased Green hub) a few times. Apologies for bothering you, as well as for the mess and my dumb and imprecise questions. This a help call after three days trying. Very many thanks in advance for any help.

**Comment by fsaris** [comment]
Step 6 isn't needed and not sure what happens when you do that.

You should adjust the example yaml to use your mesh credentials and upload that (using the esphome module in HA) to your esp32 device. See #71

I would recommend to removing awox config etc you added to your home assistant config as that isn't needed.

**Reply by MBInoob** [reply]
Very many thanks for your kind help!

Have suppressed all component's files from the config>esphome folder directory.

Apologies, but not so sure how to upload the adjusted .yaml example file to the device using ESPhome module in HA (despite googling it). What readings would you please recommend?

**Reply by fsaris** [reply]
See for instance https://esphome.io/guides/getting_started_hassio.html

**Reply by MBInoob** [reply]
Thanks! Tried editing the esp32 module .yaml file, simply pasting the Mesh password adjusted AwoX Ble Mesh code at the bottom:

```
substitutions:
  name: esphome-web-ce4d10
  friendly_name: "ESPHome Web ce4d10"

esphome:
  name: ${name}
  friendly_name: ${friendly_name}
  min_version: 2024.6.0
  name_add_mac_suffix: false
  project:
    name: esphome.web
    version: dev

esp32:
  board: esp32dev
  framework:
    type: arduino

# Enable logging
logger:

# Enable Home Assistant API
api:

# Allow Over-The-Air updates
ota:
- platform: esphome

# Allow provisioning Wi-Fi via serial
improv_serial:

wifi:
  # Set up a wifi access point
  ap: {}

# In combination with the `ap` this allows the user
# to provision wifi credentials to the device via WiFi AP.
captive_portal:

dashboard_import:
  package_import_url: github://esphome/example-configs/esphome-web/esp32.yaml@main
  import_full_config: true

# Sets up Bluetooth LE (Only on ESP32) to allow the user
# to provision wifi credentials to the device.
esp32_improv:
  authorizer: none

# To have a "next url" for improv serial
web_server:

esphome:
  name: awox-ble-mesh-hub
  friendly_name: "AwoX BLE mesh hub"

esp32:
  # Match this with the board your using
  board: nodemcu-32s
  framework:
    # Be sure to "Clean Build Files" when switching framework
    type: esp-idf

status_led:
  # Match this with the board your using
  pin: GPIO2

# Enable logging
logger:
  #level: VERY_VERBOSE

# debug:
#   update_interval: 5s

# text_sensor:
#   - platform: debug
#     reset_reason:
#      name: "Reset Reason"

# sensor:
#   - platform: debug
#     free:
#       name: "Heap Free"
#     loop_time:
#       name: "Loop Time"

mqtt:
  broker: !secret mqtt_host
  username: !secret mqtt_user
  password: !secret mqtt_password

# Disable Home Assistant API to prevent frequent MQTT disconnections
#api:

ota:
  password: !secret ota_password

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

external_components:
  - source:
      type: git
      url: https://github.com/fsaris/EspHome-AwoX-BLE-mesh-hub
      ref: main
  #  ref: dev
  # - source:
  #     type: local
  #     path: components

esp32_ble_tracker:
  scan_parameters:
    active: false

awox_mesh:
  mesh_name: !secret mesh_name
  mesh_password: !secret mesh_password

  # Optional options

  # address_prefix: A4:C1
  # min_rssi: -90 # to prevent that the mesh tries to connect to devices that are out of range
  # max_connections: 2 # max 3 connections

  # You can give a list of mesh_ids that only should be handled by this device
  # Example when you want to setup multiple ESP modules to only control a part of the mesh) be sure to also set allowed_mac_addresses option.

  # allowed_mesh_ids:
  #   - 12345
  #   - 23456
  #   - 6758

  # You can give a list of mac_addresses that are allowed to connect to (to exclude some devices that are often offline or for the allowed_mesh_ids option)

  # allowed_mac_addresses:
  #   - A4:C1:38:8A:BA:11
  #   - A4:C1:38:65:AD:22

  device_info:
    # Example device type, see device info in HomeAssistant or MQTT message to find the 'product_id'
    - product_id: 0x32
      device_type: RGB
      manufacturer: EGLO
      name: Spot 120
      model: EGLOSpot 120/w
      icon: mdi:wall-sconce-flat
```

Errors came up (two different names didn't please the editor for a start) and I wasn't sure what to do, so I tried something simpler, just adding that switch example from "[getting started hassio](https://esphome.io/guides/getting_started_hassio.html)":

```
switch:
  - platform: gpio
    name: "Living Room Dehumidifier"
    pin: GPIO5
```

I know this isn't the AwoX Ble mesh setup, but no error turned up though it didn't add a visible switch anywhere, not that I can see. Perhaps I need to get help from poeple who like to take noobs by the hand to get started. Where would I please find such poeple, if you have any idea?

**Reply by MBInoob** [reply]
For the record, never worked. Stopped trying because of the lack of clear explaination. Thanks for the help provided anyway.
